From 76b525f745de362c3a150d92181c3db215f1065d Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Thu, 13 Aug 2015 14:50:03 +0200
Subject: [PATCH 09/12] Fix authentication error notification click handling.

The activity resolved by the implicit intent is not exported by the
Email app, so convert the intent to an explicit one pointing at the
exported entry point.

Change-Id: I6f3ba590680618a0c56d1cb9786767e6a8d94a0e
---
 .../android/exchange/service/AbstractSyncAdapterService.java  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/exchange/service/AbstractSyncAdapterService.java b/src/com/android/exchange/service/AbstractSyncAdapterService.java
index 74626e95..4ac1cdc5 100644
--- a/src/com/android/exchange/service/AbstractSyncAdapterService.java
+++ b/src/com/android/exchange/service/AbstractSyncAdapterService.java
@@ -118,7 +118,9 @@ public abstract class AbstractSyncAdapterService extends Service {
                 IntentUtilities.PATH_SETTINGS);
         IntentUtilities.setAccountId(builder, accountId);
         IntentUtilities.setAccountName(builder, accountName);
-        return new Intent(Intent.ACTION_EDIT, builder.build());
+        ComponentName setupPrefComponent = new ComponentName("com.android.email",
+                "com.android.mail.ui.settings.PublicPreferenceActivity");
+        return new Intent(Intent.ACTION_EDIT, builder.build()).setComponent(setupPrefComponent);
     }
 
     protected void showAuthNotification(long accountId, String accountName) {
-- 
2.17.1

